### ISSUE TITLE
Tweak automatic variable declaration

### ISSUE THREAD

**Issue by JacquesCarette** [JacquesCarette/Drasil#495]
Context: In GlassBR, most variables are inferred, but some need to be explicitly declared.  This inconsistency should be explained (at a minimum) and, at best, fixed.

Steven mentions:
> The automatic variable declarations happen in genFunc (Code/Imperative/Import.hs:536) -- on lines 542-545, it creates a new block from the original block of function statements.  fstdecl (CodeSpec.hs:262) is called, which makes a set of all of the variables used in the functions, and subtracts from it the set of all the variables that are actually declared.  Using the output of fstdecl, the new block makes a set of variable assignment statements to all undeclared variables first and then appends the old statement block.

**Comment by JacquesCarette** [comment]
@bmaclach can you verify whether this is still a 'live' issue?

**Comment by bmaclach** [comment]
Wasn't aware of this issue, but I did notice and fix this rather recently (#2073).

The reason some variables needed to be explicitly declared was because they really needed to be initialized, and the generator was (incorrectly) set up to initialize a list variable as an empty list if it was declared using `FDec`, instead of just declaring it.

I've removed the `FDec` node altogether, and replaced it with `FDecDef` to allow for explicitly declaring and initializing in the same statement. So, while the user may still use `FDecDef` to explicitly declare things, it is now a choice and not a need.
